From f9fc74d20e3696ce0f212af05233dbb9bbc76bcd Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Fri, 3 Sep 2021 13:36:05 +0200
Subject: [PATCH 134/841] feat(stm32mp2): get SoC part number from OTP

Add the different part numbers available on STM32MP25 SoC.
And the code that allows the access to the part number OTP.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 433a1f74ad8e29e1ceed28e353e8365be0d5fc0e feat(stm32mp2): get SoC part number from OTP]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Change-Id: Ic58f834602690c046bfd29ab6ebf1917b419ef42
---
 plat/st/stm32mp2/stm32mp2_private.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/plat/st/stm32mp2/stm32mp2_private.c b/plat/st/stm32mp2/stm32mp2_private.c
index ba69d686a..b086169bf 100644
--- a/plat/st/stm32mp2/stm32mp2_private.c
+++ b/plat/st/stm32mp2/stm32mp2_private.c
@@ -140,7 +140,17 @@ uint32_t stm32mp_get_chip_dev_id(void)
 
 static uint32_t get_part_number(void)
 {
-	return 0U;
+	static uint32_t part_number;
+
+	if (part_number != 0U) {
+		return part_number;
+	}
+
+	if (stm32_get_otp_value(PART_NUMBER_OTP, &part_number) != 0) {
+		panic();
+	}
+
+	return part_number;
 }
 
 void stm32mp_get_soc_name(char name[STM32_SOC_NAME_SIZE])
-- 
2.47.2

